src/FunderMaps.Core.Authentication/SignInService.cs

```diff
@@ -0,0 +1,226 @@
+﻿using FunderMaps.Core.Entities;
+using FunderMaps.Core.Identity;
+using FunderMaps.Core.Interfaces;
+using FunderMaps.Core.Interfaces.Repositories;
+using FunderMaps.Core.Types;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace FunderMaps.Core.Authentication
+{
+    /// <summary>
+    ///     Provides the APIs for authentication.
+    /// </summary>
+    public class SignInService
+    {
+        /// <summary>
+        ///     The <see cref="IUserRepository"/> used.
+        /// </summary>
+        public IUserRepository UserRepository { get; }
+
+        /// <summary>
+        ///     The <see cref="IOrganizationUserRepository"/> used.
+        /// </summary>
+        public IOrganizationUserRepository OrganizationUserRepository { get; }
+
+        /// <summary>
+        ///     The <see cref="IOrganizationRepository"/> used.
+        /// </summary>
+        public IOrganizationRepository OrganizationRepository { get; }
+
+        /// <summary>
+        ///     The <see cref="IPasswordHasher"/> used.
+        /// </summary>
+        public IPasswordHasher PasswordHasher { get; }
+
+        /// <summary>
+        ///     Gets the <see cref="ILogger"/> used to log messages.
+        /// </summary>
+        protected ILogger Logger { get; }
+
+        /// <summary>
+        ///     Creates a new instance.
+        /// </summary>
+        public SignInService(
+            IUserRepository userRepository,
+            IOrganizationUserRepository organizationUserRepository,
+            IOrganizationRepository organizationRepository,
+            IPasswordHasher passwordHasher,
+            ILogger<SignInService> logger)
+        {
+            UserRepository = userRepository ?? throw new ArgumentNullException(nameof(userRepository));
+            OrganizationUserRepository = organizationUserRepository ?? throw new ArgumentNullException(nameof(organizationUserRepository));
+            OrganizationRepository = organizationRepository ?? throw new ArgumentNullException(nameof(organizationRepository));
+            PasswordHasher = passwordHasher ?? throw new ArgumentNullException(nameof(passwordHasher));
+            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <summary>
+        ///     Returns a flag indicating whether the specified user can sign in.
+        /// </summary>
+        /// <param name="id">The user id whose sign-in status should be returned.</param>
+        /// <returns><c>True</c> if the specified user can sign-in, otherwise false.</returns>
+        public virtual async Task<SignInContext> CanSignInAsync(Guid id)
+        {
+            if (await UserRepository.IsLockedOutAsync(id))
+            {
+                Logger.LogWarning(3, $"User {id} is currently locked out.");
```

Comment:
Are these event id's specified somewhere? I suggest not hard coding them like this.